### ISSUE TITLE
[Regression] Initial textLayer rendering not correct after PR 15722

### ISSUE THREAD

**Issue by Snuffleupagus** [mozilla/pdf.js#15795]
Attach (recommended) or Link to PDF file here:

Configuration:
- Web browser and its version: N/A
- Operating system and its version: N/A
- PDF.js version: Any version after PR #15722
- Is a browser extension: N/A

Steps to reproduce the problem:
After PR #15722, the textLayer will no longer have correct dimensions *unless* zooming has occurred at least once. This was found when testing the "pageviewer" component-example locally, see https://github.com/mozilla/pdf.js/tree/master/examples/components

To more easily reproduce this directly in the default viewer, apply the following diff and load http://localhost:8888/web/viewer.html#disableHistory=true&textLayer=visible:
```diff
diff --git a/web/app.js b/web/app.js
index e79ba7b5c..2d37c9319 100644
--- a/web/app.js
+++ b/web/app.js
@@ -1638,8 +1638,9 @@ const PDFViewerApplication = {
     if (!this.pdfViewer.currentScaleValue) {
       // Scale was not initialized: invalid bookmark or scale was not specified
.
       // Setting the default one.
-      this.pdfViewer.currentScaleValue = DEFAULT_SCALE_VALUE;
+      // this.pdfViewer.currentScaleValue = DEFAULT_SCALE_VALUE;
     }
+    this.pdfViewer.update();
   },
```

What is the expected behavior? (add screenshot)
Correct initial textLayer dimensions, as in:

![working](https://user-images.githubusercontent.com/2692120/206423655-7c4076fe-fc13-4467-b6ba-de87ab44b303.png)

What went wrong? (add screenshot)
Wrong initial textLayer dimensions, as in:

![broken](https://user-images.githubusercontent.com/2692120/206423994-9b0ec4bb-d3f7-4651-8b7b-fae018f63034.png)

Link to a viewer (if hosted on a site other than mozilla.github.io/pdf.js or as Firefox/Chrome extension): N/A

/cc @calixteman

**Comment by Snuffleupagus** [comment]
One way to address this may be to always set the following by default in the `PDFViewer`/`PDFPageView` constructors, so if that makes sense overall I can submit a patch!
```javascript
docStyle.setProperty("--scale-factor", PixelsPerInch.PDF_TO_CSS_UNITS);
```

**Comment by calixteman** [comment]
Yep it's what I found too (i.e. the fact that the scale-factor is wrong).
I'd say that this bug just highlights something which was already wrong.
Let's go for your fix :).
